components/request-shape/src/lib.rs

```diff
@@ -57,42 +57,52 @@ fn check_url(req: &IncomingRequest) -> anyhow::Result<()> {
 
 /// Check that the headers are as expected
 fn check_headers(req: &IncomingRequest) -> anyhow::Result<()> {
-    let expected_headers = [
+    let expected_headers: [(&str, &[&str]); 8] = [
         (
             "spin-raw-component-route",
-            "/base/:path_segment/:path_end/...",
+            &["/base/:path_segment/:path_end/..."],
         ),
         (
             "spin-full-url",
-            "http://example.com/base/path/end/rest?key=value",
+            &[
+                "http://example.com/base/path/end/rest?key=value",
+                "https://example.com/base/path/end/rest?key=value",
+            ],
         ),
-        ("spin-path-info", "/rest"),
+        ("spin-path-info", &["/rest"]),
         // Hardcoded base path for backwards compatibility
-        ("spin-base-path", "/"),
-        ("spin-component-route", "/base/:path_segment/:path_end"),
-        ("spin-path-match-path-segment", "path"),
-        ("spin-path-match-path-end", "end"),
-        ("spin-matched-route", "/base/:path_segment/:path_end/..."),
+        ("spin-base-path", &["/"]),
+        ("spin-component-route", &["/base/:path_segment/:path_end"]),
+        ("spin-path-match-path-segment", &["path"]),
+        ("spin-path-match-path-end", &["end"]),
+        ("spin-matched-route", &["/base/:path_segment/:path_end/..."]),
     ];
 
     let mut actual_headers: HashMap<String, Vec<Vec<u8>>> = HashMap::new();
     for (k, v) in req.headers().entries() {
         actual_headers.entry(k).or_default().push(v);
     }
 
-    for (name, value) in expected_headers.into_iter() {
+    for (name, values) in expected_headers.into_iter() {
         let header = header_as_string(&mut actual_headers, name)?;
 
-        anyhow::ensure!(
-            header == value,
-            "Header {name} was expected to contain value '{value}' but contained '{header}' "
-        );
+        for (i, value) in values.iter().enumerate() {
+            if header != *value && i == values.len() - 1 {
+                anyhow::bail!(
+                    "Header {name} was expected to contain value '{value}' but contained '{header}' "
+                );
+            }
+        }
```

Comment:
Assuming `values` is a set of "acceptable" values I don't think this code is correct for multiple `values`.

Perhaps:
```suggestion
        anyhow::ensure!(
          values.contains(&header),
          ""Header {name} value was expected to be one of {values:?} but was '{header}'",
        );
```

Reply:
Wow - yea, not sure what I wrote there...